test/celeritas/ext/GeantImporter.test.cc

```diff
@@ -1293,6 +1309,89 @@ TEST_F(OneSteelSphereGG, physics)
+    EXPECT_EQ(1, imported.optical.size());
+
+    // First material is vacuum, no optical properties
+    MaterialId vacuum_id{0};
+    EXPECT_EQ("vacuum", imported.materials[vacuum_id.get()].name);
+    auto const vacuum_iter = imported.optical.find(vacuum_id.get());
+    EXPECT_TRUE(vacuum_iter == imported.optical.end());
+
+    // Second material is liquid argon
+    MaterialId lar_id{1};
+    EXPECT_EQ("lAr", imported.materials[lar_id.get()].name);
+    auto const lar_iter = imported.optical.find(lar_id.get());
+    ASSERT_FALSE(lar_iter == imported.optical.end());
+    auto const& optical = lar_iter->second;
+
+    // Check optical material ID
+    auto materials = MaterialParams::from_import(imported);
+    ASSERT_TRUE(materials);
+    EXPECT_EQ(OpticalMaterialId{},
+              materials->get(vacuum_id).optical_material_id());
+    EXPECT_EQ(OpticalMaterialId{0},
+              materials->get(lar_id).optical_material_id());
+
+    real_type const tol = this->comparison_tolerance();
+
+    // Check scintillation optical properties
+    auto const& scint = optical.scintillation;
+    EXPECT_TRUE(scint);
+    EXPECT_EQ(1, scint.resolution_scale);
+    EXPECT_EQ(5, scint.yield);
+    EXPECT_EQ(3, scint.components.size());
+    std::vector<double> components;
+    for (auto const& comp : scint.components)
+    {
+        components.push_back(comp.yield);
+        components.push_back(to_cm(comp.lambda_mean));
+        components.push_back(to_cm(comp.lambda_sigma));
+        components.push_back(to_sec(comp.rise_time));
+        components.push_back(to_sec(comp.fall_time));
+    }
+    // clang-format off
+    static double const expected_components[]
+        = {3, 1.28e-05, 1e-06, 1e-08, 6e-09,
+           1, 1.28e-05, 1e-06, 1e-08, 1.5e-06,
+           1, 2e-05,    2e-06, 1e-08, 3e-06};
+    // clang-format on
+    EXPECT_VEC_NEAR(expected_components, components, tol);
+
+    // Check Rayleigh optical properties
+    auto const& rayleigh = optical.rayleigh;
+    EXPECT_TRUE(rayleigh);
+    EXPECT_EQ(1, rayleigh.scale_factor);
+    EXPECT_REAL_EQ(0.024673059861887867,
+                   rayleigh.compressibility * units::gram
+                       / (units::centimeter * units::second * units::second));
+    EXPECT_EQ(11, rayleigh.mfp.x.size());
+    EXPECT_DOUBLE_EQ(1.55e-06, rayleigh.mfp.x.front());
+    EXPECT_DOUBLE_EQ(1.55e-05, rayleigh.mfp.x.back());
+    EXPECT_REAL_EQ(32142.9, to_cm(rayleigh.mfp.y.front()));
+    EXPECT_REAL_EQ(54.6429, to_cm(rayleigh.mfp.y.back()));
+
+    // Check absorption optical properties
+    auto const& abs = optical.absorption;
+    EXPECT_TRUE(abs);
+    EXPECT_EQ(2, abs.absorption_length.x.size());
+    EXPECT_DOUBLE_EQ(1.3778e-06, abs.absorption_length.x.front());
+    EXPECT_DOUBLE_EQ(1.55e-05, abs.absorption_length.x.back());
+    EXPECT_REAL_EQ(86.4473, to_cm(abs.absorption_length.y.front()));
+    EXPECT_REAL_EQ(0.000296154, to_cm(abs.absorption_length.y.back()));
```

Comment:
Same here. Seems too short?

Reply:
From the same example, except I only used the endpoints in the test gdml instead of copying the whole absorption length vector. Since we're only using the input to test that the values are imported correctly right now I wasn't too worried about the accuracy, but it would be good to have realistic values if possible.